### PR TITLE
unittests: Fixes unit tests for Windows (part 12)

### PR DESCRIPTION
**Pull request by claudiubelu**
#### What type of PR is this?

/kind failing-test
/sig windows
/sig testing

#### What this PR does / why we need it:

Currently, there are some unit tests that are failing on Windows due to various reasons:

- IPVS proxy mode is not supported on Windows.
- ``pkg/kubelet/cri/remote`` was moved to ``cri-client``.

#### Which issue(s) this PR fixes:

Related: https://github.com/kubernetes/kubernetes/issues/51540

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by claudiubelu** [comment]
All unit tests passed.

**Review by neolit123 — COMMENTED** [review]
thanks @claudiubelu 
do we have an ETA for enabling the windows unit tests on PRs as presubmits?
looks like my recent kubeadm changes broke the tests on Windows again without knowing.
cc @jsturtevant @marosset 

/approve
for kubeadm (can't approve for /pkg)

File: pkg/proxy/apis/config/validation/validation_test.go [hunk 1]
```diff
@@ -193,7 +189,19 @@ func TestValidateKubeProxyConfiguration(t *testing.T) {
 			},
 			expectedErrs: field.ErrorList{field.Invalid(newPath.Child("logging.format"), "unsupported format", "Unsupported log format")},
 		},
-	} {
+	}
+
+	if runtime.GOOS != "windows" {
```
Comment (neolit123) [inline pkg/proxy/apis/config/validation/validation_test.go:194]: can this be organized with go build tags / separate files instead?

Comment (claudiubelu) [reply pkg/proxy/apis/config/validation/validation_test.go]: Done

**Comment by claudiubelu** [comment]
> thanks @claudiubelu do we have an ETA for enabling the windows unit tests on PRs as presubmits? looks like my recent kubeadm changes broke the tests on Windows again without knowing. cc @jsturtevant @marosset
> 
> /approve for kubeadm (can't approve for /pkg)

I have sent a PR for the presubmit job here: https://github.com/kubernetes/test-infra/pull/32689 the plan is to have it as informing at the moment.

**Comment by neolit123** [comment]
> > thanks @claudiubelu do we have an ETA for enabling the windows unit tests on PRs as presubmits? looks like my recent kubeadm changes broke the tests on Windows again without knowing. cc @jsturtevant @marosset
> > /approve for kubeadm (can't approve for /pkg)
> 
> I have sent a PR for the presubmit job here: [kubernetes/test-infra#32689](https://github.com/kubernetes/test-infra/pull/32689) the plan is to have it as informing at the moment.

that's great. thanks!

this PR here needs update for some missing vars:
```
ERROR(linux/amd64): /home/prow/go/src/k8s.io/kubernetes/pkg/proxy/apis/config/validation/validation_linux_test.go:47:48: undefined: newPath
ERROR(linux/amd64): /home/prow/go/src/k8s.io/kubernetes/pkg/proxy/apis/config/validation/validation_linux_test.go:23:2: "runtime" imported and not used
ERROR(linux/amd64): /home/prow/go/src/k8s.io/kubernetes/pkg/proxy/apis/config/validation/validation_linux_test.go:24:2: "testing" imported and not used
ERROR(linux/amd64): /home/prow/go/src/k8s.io/kubernetes/pkg/proxy/apis/config/validation/validation_linux_test.go:25:2: "time" imported and not used
ERROR(linux/amd64): /home/prow/go/src/k8s.io/kubernetes/pkg/proxy/apis/config/validation/validation_linux_test.go:27:2: "github.com/stretchr/testify/assert" imported and not used
ERROR(linux/amd64): /home/prow/go/src/k8s.io/kubernetes/pkg/proxy/apis/config/validation/validation_linux_test.go:31:2: "k8s.io/component-base/config" imported as componentbaseconfig and not used
ERROR(linux/amd64): /home/prow/go/src/k8s.io/kubernetes/pkg/proxy/apis/config/validation/validation_linux_test.go:32:2: "k8s.io/component-base/logs/api/v1" imported as logsapi and not used
ERROR(linux/amd64): /home/prow/go/src/k8s.io/kubernetes/pkg/proxy/apis/config/validation/validation_linux_test.go:34:2: "k8s.io/utils/ptr" imported and not used
```

**Review by neolit123 — COMMENTED** [review]
/lgtm
/assign @dims 
PTAL for pkg approval if possible

**Comment by claudiubelu** [comment]
/hold

Until another windows unit tests ci run finishes.

**Comment by claudiubelu** [comment]
> /hold
> 
> Until another windows unit tests ci run finishes.

Unit tests passed. (https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/125545/pull-ci-kubernetes-unit-windows/1806342440857112576)

/hold cancel

**Review by neolit123 — COMMENTED** [review]
/lgtm

**Pull request merged**